Core/Resource/SymbolJson.cs

```diff
@@ -30,218 +30,215 @@ public static void WriteObject(this JsonTextWriter writer, string name, object v
                 }
 
-                Writer.WriteEndArray();
+                writer.WriteEndArray();
 
-                Writer.WritePropertyName("Outputs");
-                Writer.WriteStartArray();
+                writer.WritePropertyName(JsonKeys.Outputs);
+                writer.WriteStartArray();
 
                 foreach (var (id, output) in child.Outputs)
                 {
                     if (output.DirtyFlagTrigger != output.OutputDefinition.DirtyFlagTrigger || output.OutputData != null || output.IsDisabled)
                     {
-                        Writer.WriteStartObject();
-                        Writer.WriteValue("Id", id);
-                        Writer.WriteComment(child.ReadableName);
+                        writer.WriteStartObject();
+                        writer.WriteValue(JsonKeys.Id, id);
+                        writer.WriteComment(child.ReadableName);
 
                         if (output.OutputData != null)
                         {
-                            Writer.WritePropertyName("OutputData");
-                            Writer.WriteStartObject();
-                            Writer.WriteObject("Type", output.OutputData.DataType);
-                            output.OutputData.ToJson(Writer);
-                            Writer.WriteEndObject();
+                            writer.WritePropertyName(JsonKeys.OutputData);
+                            writer.WriteStartObject();
+                            writer.WriteObject(JsonKeys.Type, output.OutputData.DataType);
+                            output.OutputData.ToJson(writer);
+                            writer.WriteEndObject();
                         }
 
                         if (output.DirtyFlagTrigger != output.OutputDefinition.DirtyFlagTrigger)
                         {
-                            Writer.WriteObject("DirtyFlagTrigger", output.DirtyFlagTrigger);
+                            writer.WriteObject(JsonKeys.DirtyFlagTrigger, output.DirtyFlagTrigger);
                         }
 
                         if (output.IsDisabled)
                         {
-                            Writer.WriteValue("IsDisabled", output.IsDisabled);
+                            writer.WriteValue(JsonKeys.IsDisabled, output.IsDisabled);
                         }
 
-                        Writer.WriteEndObject();
+                        writer.WriteEndObject();
                     }
                 }
 
-                Writer.WriteEndArray();
+                writer.WriteEndArray();
 
-                Writer.WriteEndObject(); // child
+                writer.WriteEndObject(); // child
             }
 
-            Writer.WriteEndArray();
+            writer.WriteEndArray();
         }
         #endregion
 
         #region reading
-        private static SymbolChild ReadSymbolChild(Model model, JToken symbolChildJson)
+        private static bool TryReadSymbolChild(Model model, JToken symbolChildJson, out SymbolChild symbolChild)

```

Comment:
Like many other functions so far, I've also changed this to the `TryX` paradigm that has become popular in C#. It vastly reduces the number of null checks we have to perform on the results, and is straightforward about the fact that this function can fail.